gourmet/Undo.py

```diff
@@ -92,7 +94,9 @@ def __init__ (self, set_text_action, history, initial_text="",text="",txt_id=Non
             debug('Too many changes - assume 0,0',0)
             self.cindex,self.clen = 0,0
 
-    def determine_mode(self, text: str = "", initial_text: str = "") -> str:
+    def determine_mode(self,
+                       text: Optional[str] = "",
+                       initial_text: Optional[str] = "") -> str:
```

Comment:
I don't want to be a burden, but we only return `add` and `delete` and we don't have to change this here and now (we can create a new ticket and PR for this), but this looks suspiciously like add/delete it should be an `Enum`.

Reply:
otherwise I am fine with this. We just have to note this for later.

Reply:
Maybe a boolean?

Reply:
Definitely an Enum. Boolean is also an Enum but True/False is not a "Mode".

Reply:
Ok, that did uncover a bug. Can you have at it again?